### PR TITLE
Pin SciPy to `<1.8.0`

### PR DESCRIPTION
**Pull request by michaelosthege**
See tracking issue #5448.

### CODE REVIEW

**Comment by OriolAbril** [comment]
rtd is still failing, _from trying to install numpy==1.19_ which I can only assume is a strange requirement of some testing library. We might need to split test/docs... requirements for this to work (and therefore address https://github.com/pymc-devs/pymc/issues/5074)

**Comment by michaelosthege** [comment]
Running `code --diff .github\workflows\pytest.yml .github\workflows\jaxtests.yml` and `code --diff .github\workflows\pytest.yml .github\workflows\arviz_compat.yml` it looks like the environments between those workflows are quite different:
* `pytest.yml` vs. `jaxtests.yml`/`arviz-compat.yml`
* `ubuntu-18-04` vs. `ubuntu-latest`/`maxos-latest`
* `environment-dev-py37.yml` vs. `environment-dev-py39.yml`
* no extras vs. `pip install "numpyro>=0.8.0"`/`pip install git+git://github.com/arviz-devs/arviz.git`

**Comment by OriolAbril** [comment]
I guess there are significant differences between ubuntu 18 and latest (should be 20 I think). But also with python 3.7! IIRC, the latest versions of numpy/scipy... don't support it anymore, there was a NEP about this but don' t have the link at hand, I'll try to come back and help a bit more tonight.

**Comment by twiecki** [comment]
This is probably fine as a hot fix, but technically this is not a pymc issue but an aesara one and that's where it should be fixed (e.g. by pinning the version there), rather than here.

**Comment by twiecki** [comment]
Just seeing it's already fixed there: https://github.com/aesara-devs/aesara/pull/812 so shouldn't we just wait for a new aesara release and pin to that?

**Comment by twiecki** [comment]
If you pin `aesara=2.4.0` we shouldn't need to pin anything else here.

**Comment by twiecki** [comment]
Ah, that's what you did in [498dc43](https://github.com/pymc-devs/pymc/pull/5449/commits/498dc435c5c65e23a1497387f5f3a88fdbdf8e61). That actually looks right, not sure why numpy even gets compiled in the arviz installation?

**Comment by ricardoV94** [comment]
~~Just quick check to see if problems come from the recent change to mamba...~~

It wasn't that

File: conda-envs/environment-dev-py38.yml [hunk 1]
```diff
@@ -22,8 +22,7 @@ dependencies:
 - pytest-cov>=2.5
 - pytest>=3.0
 - python-graphviz
-- python=3.8
```
Comment (ricardoV94) [inline conda-envs/environment-dev-py38.yml:25]: Why was the python version dependency removed?

Comment (twiecki) [reply conda-envs/environment-dev-py38.yml:25]: Yeah that's not right.

**Comment by ricardoV94** [comment]
Okay installation seems to have worked now!

**Comment by ricardoV94** [comment]
Also weirdly... arviz compat and jax workflows are using the "dev" environments, and not the "test" ones... Why is that?

And some environments pin the pandas version but not others...

**Comment by ricardoV94** [comment]
Reverting the Aesara upgrade, because that fails due to https://github.com/pymc-devs/pymc/issues/5446

**Comment by twiecki** [comment]
> Also weirdly... arviz compat and jax workflows are using the "dev" environments, and not the "test" ones... Why is that?

It shouldn't, oversight.

**Comment by ricardoV94** [comment]
The `pip-from-conda` hook does not seem to handle double sided version limits...

**Comment by ricardoV94** [comment]
Ok, so `scipy` appears in the list of packages that should be excluded from requirements... not sure why this is:

https://github.com/pymc-devs/pymc/blob/b54c8af5c4a43b1a46b6597ebae4ddec1b5b22ca/scripts/generate_pip_deps_from_conda.py#L53

**Comment by ricardoV94** [comment]
Docs failure is related to Aeppl dependency. Will open a separate PR to update it

**Review by twiecki — APPROVED** [review]

**Pull request merged**